src/main/java/org/traccar/WebDataHandler.java

```diff
@@ -287,6 +288,15 @@ public void run(Timeout timeout) {
     @Override
     protected Position handlePosition(Position position) {
 
+        Device device = identityManager.getById(position.getDeviceId());
+
+        Map<String, Object> attributes = device.getAttributes();
+        if (attributes.containsKey(ATTRIBUTE_DEVICE_ALLOW_FORWARDING)) {
+            if (!(Boolean) attributes.get(ATTRIBUTE_DEVICE_ALLOW_FORWARDING)) {
```

Comment:
I don't think you should do a lookup twice here.

Reply:
Fixed. Doing only one get for attribute and then check if it is null or false